form/form.go

```
@@ -325,6 +326,14 @@ func intEncoder(values *Values, v reflect.Value, keyParts []string, encodeZero b
 	values.Add(FormatKey(keyParts), strconv.FormatInt(val, 10))
 }
 
+func timeEncoder(values *Values, v reflect.Value, keyParts []string, encodeZero bool, _ *formOptions) {
+	val := v.Interface().(time.Time)
```

what happens if this is called with a v other than a time.Time; does this return an error we need to look for, or will val just be Zero (and is that swallowing an error?)

---

It panics. I think that's OK, but open for other thoughts. It's only called in one location surrounded by
```
if t == reflect.TypeOf(time.Time{}) {
	return timeEncoder
}
```
so logically, the type assertion `v.Interface().(time.Time)` must succeed. I would definitely rather fail fast here rather than silently. The alternative would be to bubble up an `error`, but it feels a little weird to return an `error` due to a bug.

---

In fact, I'm realizing we have the same `panic` behavior in the rest of type assertions in `form.go`, so this is really no different than current behavior (where we do a type assertion to an integer: https://github.com/stripe/stripe-go/blob/b384f33fa0506bc052f50bf8c8be01b109278d05/form/form.go#L321

---

Cool, if its consistent with what we have I think this is good.  Maybe this is something we should look at separately, i.e. if a panic is the right response here (vs something more graceful) but that would be separate from this PR.